src/lib/cache/gdfs_cache.hpp

```diff
@@ -1,19 +1,22 @@
 #pragma once
 
 #include <list>
-#include <unordered_map>
+#include <shared_mutex>
 #include <utility>
 
-#include "abstract_cache_impl.hpp"
+#include "abstract_cache.hpp"
 #include "boost/heap/fibonacci_heap.hpp"
 #include "utils/assert.hpp"
 
 namespace opossum {
 
-// Generic cache implementation using the GDFS policy.
-// Note: This implementation is not thread-safe.
+/*
+ * Generic cache implementation using the GDFS policy.
+ * To iterate over a cache copy, use snapshot().
```

Comment:
Could clarify why I want this, e.g., "To iterate over the cache in a thread-safe manner, ..."